### DISCUSSION TITLE
filter_count wrong with _or filter

### DISCUSSION THREAD

**Discussion by jakobpevec** [directus/directus#16992] [Archive]
Hey!

I'm creating a search field, which searches title and article number fields with a conditonal filter. The expected result would be ie. 0 results, but filter_count outputs 53. Or expected result: 7, filter_count: 15.

Am i doing something wrong or is this a bug?

Here's the filter:

```javascript 
productQuery = {
            status: {
                _eq: 'published'
            },
            _or: [
                {
                    translations: {
                        product_title: {
                            _contains: query
                        }
                    }
                },
                {

                    product_variations: {
                        product_variation_id: {
                            article_number: {
                                _contains: query
                            }
                        }
                    }
                }
            ]
        };
```

Thanks for the answers in advance :)

**Comment by azrikahar** [comment]
Can you try wrapping `status: { ... }` and `_or: [ ... ]` inside an `_and` logical operator array? Although understandably you are expecting your current usage to imply the `AND` logic, but it is not really a valid filter, at least as of now. Additional reference: https://docs.directus.io/reference/filter-rules.html#logical-operators

**Reply by jakobpevec** [reply]
Aha, I wasn't sure if that was the case, thanks. The filter_count still has the same problem though. For now I'm solving this with an additional aggregate count query.

Comparison:

```javascript 
result:
{ filter_count: 68 }
{ aggregateCount: 3 }


filter query:

_and: [
                {
                    status: {
                        _eq: 'published'
                    },
                },
                {
                    _or: [
                        {
                            translations: {
                                product_title: {
                                    _contains: query
                                }
                            }
                        },
                        {

                            product_variations: {
                                product_variation_id: {
                                    article_number: {
                                        _contains: query
                                    }
                                }
                            }
                        }
                    ]
                }
            ]
```

**Reply by azrikahar** [reply]
Thanks for confirming!

> The filter_count still has the same problem though. For now I'm solving this with an additional aggregate count query.

Hmm can you share what Directus version this is? If you are on `9.22.1`, can you try `9.21.1` to see are there any difference? And if you are on `9.21.1` instead, can you try `9.22.1`?

---

On a slight tangent, aggregate count query will technically be the recommended approach in the future when #15665 gets implemented, so it is still very helpful to know aggregate count works in the mean time 👍

**Reply by jakobpevec** [reply]
I upgraded from 9.21.1 to 9.22.1 yesterday, I had the same issue on both versions.

Aggregate count also gives the same wrong count, but countDistinct gives the correct count :)

**Reply by mlbiche** [reply]
Same problem here, with the latest Directus version. Even in the Directus application, I do have the issue : the `filter_count` states 13, but the data list contains 10 items. And it is the same with other collections.

The `countDistinct` aggregation states the correct count : 10.

My Directus is running above a Postgres DB.
